src/couch/src/couch_file.erl

```diff
@@ -535,13 +512,13 @@ handle_call({pread_iolists, PosL}, _From, File) ->
     ),
     Resps = read_multi_raw_iolists_int(File, LocNums2),
     Extracted = lists:zipwith(
-        fun({LenIoList, _}, {IoList, _}) ->
+        fun({LenIoList, _}, {FullIoList, _}) ->
             case iolist_to_binary(LenIoList) of
                 <<1:1/integer, _:31/integer>> ->
-                    {Checksum, IoList} = extract_checksum(IoList),
+                    {Checksum, IoList} = extract_checksum(FullIoList),
```

Comment:
How about updating the order of elements in the result tuple of `extract_checksum`? So we return `{IoList, Checksum}`. This function only used two times.

Reply:
The bag with matching on the `IoList` is a nice catch!!!

Reply:
Yeah we could swap it, that might make it more clear we're expecting a new, different IoList as a result.

Reply:
Even better swapping `IoList` and `Checksum` lets us skip the extra line in `case` cause as we can directly return `extract_checksum(FullIoList)` result.

Reply:
Updated! It saved a few extra case clause lines. Thanks for the suggestion, Ilya!